test/jasmine/tests/plot_interact_test.js

```diff
@@ -142,6 +150,119 @@ describe('Test plot structure', function() {
             });
         });
 
+        describe('scatter drag', function() {
+
+            var mock = require('@mocks/scatter_simple.json'),
+                gd, modeBar, relayoutCallback;
+
+            beforeEach(function(done) {
+                gd = createGraphDiv();
+
+                Plotly.plot(gd, mock.data, mock.layout).then(function() {
+
+                    modeBar = gd._fullLayout._modeBar;
+                    relayoutCallback = jasmine.createSpy('relayoutCallback');
+
+                    gd.on('plotly_relayout', relayoutCallback);
+
+                    delay(done);
+                });
+            });
+
+            it('scatter plot should respond to drag interactions', function(done) {
+
+                jasmine.addMatchers(customMatchers);
+
+                var precision = 5;
+
+                var buttonPan = selectButton(modeBar, 'pan2d');
+
+                var originalX = [-0.3169014084507042, 5.316901408450704];
+                var originalY = [-0.5806379476536665, 6.218528262566369];
+
+                expect(gd.layout.xaxis.range).toBeCloseToArray(originalX, precision);
+                expect(gd.layout.yaxis.range).toBeCloseToArray(originalY, precision);
+
+                // Switch to pan mode
+                expect(buttonPan.isActive()).toBe(false); // initially, zoom is active
+                buttonPan.click();
+                expect(buttonPan.isActive()).toBe(true); // switched on dragmode
+
+                // Switching mode must not change visible range
+                expect(gd.layout.xaxis.range).toBeCloseToArray(originalX, precision);
+                expect(gd.layout.yaxis.range).toBeCloseToArray(originalY, precision);
+
+                setTimeout(function() {
+
+                    expect(relayoutCallback).toHaveBeenCalledTimes(1);
+                    relayoutCallback.calls.reset();
+
+                    // Drag scene along the X axis
+
+                    mouseEvent('mousedown', 110, 150);
+                    mouseEvent('mousemove', 220, 150);
+                    mouseEvent('mouseup', 220, 150);
+
```

Comment:
> maybe needing more than the clientX / clientY coordinates.

Not sure if you've tried that yet, but here's how I made the [`plotly_click`  test](https://github.com/plotly/plotly.js/blob/d8e27dd33270aa7e4bb01a612cc99c51fa3cb68a/test/jasmine/tests/gl_plot_interact_test.js#L135) work in gl3d.